File: BarkingDog/5_stack/10828.py
#10828 .스택
"""
기초 스택 문제
"""

import sys
input = sys.stdin.readline

stack = []

N=int(input().rstrip())

for _ in range(N):
    order= input().rstrip().split()
    if order[0] == "push":
        stack.append(int(order[1]))
    elif order[0] == "top":
        print(stack[-1] if stack else -1)
    elif order[0] == "size":
        print(len(stack))
    elif order[0] == "empty":
        print(int(bool(not stack)))
    elif order[0] == "pop":
        if stack: print(stack.pop())
        else: print(-1)

File: BarkingDog/5_stack/10828.cpp
/*
10828. 스택

입출력 이슈 잘 지키자!~



*/



//N 초기화 잘 시키자!


#include <bits/stdc++.h>
using namespace std;


int main(void) {
	ios::sync_with_stdio(0);
	cin.tie(0);
	int N;
	stack <int> S;
	cin >> N ;
	for (int i=0;i<N;i++){
		string order;
		cin >>order;
		if (order == "push"){
			int num;
			cin >>num;
			S.push(num);
		}
		else if(order == "pop"){
			if(S.empty()){cout<<-1<<'\n';}
			else{cout <<S.top()<<"\n";
				S.pop();}
		}
		else if (order == "size"){
			cout <<S.size()<<"\n";
		}

		else if (order == "empty"){
			if(S.empty()){cout<<1<<"\n";}
			else {cout<<0<<"\n";}
		}
		else if(order=="top"){
			if(S.empty()){cout<<-1<<"\n";}
			else {cout<<S.top()<<"\n";}
			}

	}
	
	return 0;	
}